.github/workflows/ruff-check.yml

```
@@ -13,52 +13,66 @@ permissions:
+          
+          # Convert multiline string to space-separated for ruff
+          FILES_ARGS=$(echo "${{ steps.changed-files.outputs.files }}" | tr '\n' ' ')
+          ruff check $FILES_ARGS
+      
       # 5. Skip Ruff when no relevant files changed
       - name: Skip Ruff if no files changed
-        if: steps.changed-files.outputs.files == ''
+        if: steps.changed-files.outputs.has_files == 'false'
         run: |
-          echo "No Python files changed in api-server/ or state-manager/. Skipping Ruff."
+          echo "No Python files changed. Skipping Ruff."
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)